lib/ui/views/team/subpages/team-results.dart

```diff
@@ -95,31 +103,36 @@ class TeamResults extends StatelessWidget {
       child: Card(
         child: Column(
           crossAxisAlignment: CrossAxisAlignment.start,
-          children: card
-        )
-      )
+          children: card,
+        ),
+      ),
     );
   }
 
   Future<List<TeamParticipant>> getTeamParticipants(String teamKey) async {
     List<TeamParticipant> teamParticipants = [];
 
     // Get all the team's rankings
-    List<Ranking> allRankings = await ApiV3().getTeamResults(teamKey, StaticData().seasonKey);
+    List<Ranking> allRankings = await ApiV3().getTeamResults(teamKey, StaticData.seasonKey);
+
+    final teamEvents = await ApiV3().getTeamEvents(teamKey, StaticData.seasonKey);
 
-    await ApiV3().getTeamEvents(teamKey, StaticData().seasonKey).then((events) async {
-      events.sort(Sort().eventParticipantSorter);
+    try {
+      teamEvents.sort(Sort().eventParticipantSorter);
 
-      for (int i = 0; i < events.length; i++) {
-        EventParticipant eventParticipant = events[i];
+      List<Future<void>> requests = [];
+
+      for (final event in teamEvents) {
+        EventParticipant eventParticipant = event;
         TeamParticipant teamParticipant = TeamParticipant();
 
         // Get event detail
-        teamParticipant.event = await ApiV3().getEvent(eventParticipant.eventKey);
+        teamParticipant.event =
+            await ApiV3().getEvent(eventParticipant.eventKey);
 
         // Get team matches
-        List<Match> teamMatches = [];
-        await ApiV3().getEventMatches(eventParticipant.eventKey).then((matches) {
+        requests.add(ApiV3().getEventMatches(eventParticipant.eventKey).then((matches) {
+          List<Match> teamMatches = [];
           for (int i = 0; i < matches.length; i++) {
             Match match = matches[i];
             for (int i = 0; i < match.participants.length; i++) {
```

Comment:
Can we rewrite it with `.where`?

Reply:
definitely, I'm surprised I missed that!